### PR TITLE
[Console] increased Command test code coverage + minor improvements

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
| Q             | A
| ------------- | ---
| Branch?       | "master"
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

- added 6 test cases to ``CommandTest.php`` and improved an existing one
- added getter method for ``$processTitle`` in ``Command.php``

The code coverage of the Command class has been improved from 86.32% to 96.61%

before:
before

after:
after

### CODE REVIEW

File: src/Symfony/Component/Console/Command/Command.php [hunk 1]
```diff
@@ -198,11 +198,12 @@ protected function initialize(InputInterface $input, OutputInterface $output)
      * setCode() method or by overriding the execute() method
      * in a sub-class.
      *
-     * @param InputInterface  $input  An InputInterface instance
+     * @param InputInterface $input An InputInterface instance
      * @param OutputInterface $output An OutputInterface instance
      *
      * @return int The command exit code
      *
+     * @throws \Exception
```
Comment (stof) [inline src/Symfony/Component/Console/Command/Command.php]: this does not make any sense

File: src/Symfony/Component/Console/Command/Command.php [hunk 2]
```diff
@@ -436,13 +447,13 @@ public function setProcessTitle($title)
     }
 
     /**
-     * Returns the command name.
+     * Returns the process title of the command.
      *
      * @return string The command name
      */
-    public function getName()
+    public function getProcessTitle()
```
Comment (stof) [inline src/Symfony/Component/Console/Command/Command.php]: please don't move methods around. It does not provide any value and makes the history harder to use due to meaningless changes

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php [hunk 3]
```diff
@@ -98,6 +121,8 @@ public function testGetNamespaceGetNameSetName()
 
     /**
      * @dataProvider provideInvalidCommandNames
+     *
+     * @param $name
```
Comment (stof) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: does not make any sense (and the phpdoc is not even valid)

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php [hunk 4]
```diff
@@ -319,6 +368,9 @@ public function getSetCodeBindToClosureTests()
 
     /**
      * @dataProvider getSetCodeBindToClosureTests
+     *
+     * @param $previouslyBound
+     * @param $expected
```
Comment (stof) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: same here

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php [hunk 5]
```diff
@@ -344,6 +396,10 @@ public function testSetCodeWithNonClosureCallable()
         $this->assertEquals('interact called'.PHP_EOL.'from the code...'.PHP_EOL, $tester->getDisplay());
     }
 
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
```
Comment (stof) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: this does not provide any value. The method signature already gives all info.

**Comment by stof** [comment]
and -1 for adding a new method in the Command class to get the process title.

**Comment by ShinDarth** [comment]
@stof how can I get the process title without adding a getter for it?

**Comment by ShinDarth** [comment]
@stof corrected all the other issues, waiting for an answer about the process title

**Comment by jbafford** [comment]
@ShinDarth A better way to test `setProcessTitle()` would be to ensure that setting the title has the desired effect. That is, if you set the title, does the process generated in `run()` actually have that title? That way, you're testing the functionality and the reason *why* `setProcessTitle()` exists, and not just that it set a private internal variable.

**Comment by nicolas-grekas** [comment]
Using [cli_get_process_title](http://php.net/manual/en/function.cli-get-process-title.php)? 
The function is always available since 5.5 cli (which matches or requirements when running the test suite).

**Comment by nicolas-grekas** [comment]
I guess we should merge this on a lower branch. 2.7?

**Comment by ShinDarth** [comment]
@nicolas-grekas then should I rebase to 2.7 and modify the target branch?

**Comment by nicolas-grekas** [comment]
Would be nice to try yes

**Comment by ShinDarth** [comment]
ok I will, currently working on your suggestion

**Comment by ShinDarth** [comment]
@nicolas-grekas 

few notes:

- I wasn't able to rebase because of many merge conflicts, so I opened a new one [here](https://github.com/symfony/symfony/pull/20817)
- in the 2.7 there is no ``TestCommand::setHidden()`` method, so that should be added to another branch
- also in 2.7 branch there is not  ``Symfony\Component\Console\Exception\InvalidArgumentException`` so instead I had to use ``\InvalidArgumentException``, that should be changed as well in next branches

**Comment by nicolas-grekas** [comment]
@ShinDarth now that #20817 is merged, could you try rebasing/opening the remaining on 2.8, or maybe 3.1/3.2 depending on where each part applies?

**Comment by ShinDarth** [comment]
@nicolas-grekas yes I will do it

**Comment by ShinDarth** [comment]
@nicolas-grekas done, see: https://github.com/symfony/symfony/pull/21067

**Pull request closed without merging**